src/test/java/org/generationcp/middleware/dao/AttributeDAOTest.java

```diff
@@ -2,74 +2,86 @@
 
 import com.google.common.collect.Lists;
 import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.GermplasmTestDataGenerator;
 import org.generationcp.middleware.IntegrationTestBase;
 import org.generationcp.middleware.api.brapi.v1.attribute.AttributeDTO;
-import org.generationcp.middleware.dao.oms.CVTermDao;
+import org.generationcp.middleware.api.brapi.v2.attribute.AttributeValueDto;
 import org.generationcp.middleware.data.initializer.GermplasmTestDataInitializer;
 import org.generationcp.middleware.domain.oms.CvId;
+import org.generationcp.middleware.domain.search_request.brapi.v2.AttributeValueSearchRequestDto;
+import org.generationcp.middleware.manager.DaoFactory;
 import org.generationcp.middleware.pojos.Attribute;
+import org.generationcp.middleware.pojos.AttributeExternalReference;
 import org.generationcp.middleware.pojos.Germplasm;
 import org.generationcp.middleware.pojos.oms.CVTerm;
+import org.generationcp.middleware.pojos.workbench.CropType;
 import org.hibernate.Session;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.data.domain.Pageable;
 import org.springframework.util.CollectionUtils;
 
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
-
 public class AttributeDAOTest extends IntegrationTestBase {
-	
-	private AttributeDAO attributeDao;
-	private CVTermDao cvTermDao;
-	private GermplasmDAO germplasmDao;
+
 	private List<Integer> gids;
-	private List<String> guids;
+	private List<Germplasm> germplasms;
 	private List<CVTerm> testAttributeTypes;
 	private Attribute attribute1;
 	private Attribute attribute2;
 	private Attribute attribute3;
-	
+
+	private static final String NOTE_ATTRIBUTE = "NOTE_AA_text";
+	private CropType cropType;
+
+	private DaoFactory daoFactory;
+
+	private GermplasmTestDataGenerator germplasmTestDataGenerator;
+	private Pageable pageable;
```

Comment:
Instead of mocking Pageable interface, you can replace test method usages as: `new PageRequest(0, 10)`. It is less costly to instantiate real objects versus using mocks